# Project title    

## Author: Chang Zhang 
    
### Ecological Genomics:   

## Overall Description of notebook      

fill in your description here!


## Date started: (2017-01-18)   
## Date end:   (year-month-day)    

## Philosophy   
Science should be reproducible and one of the best ways to achieve this is by logging research activities in a notebook. Because science/biology has increasingly become computational, it is easier to document computational projects in an electronic form, which can be shared online through Github.    

### Helpful features of the notebook     

**It is absolutely critical for your future self and others to follow your work.**     

* The notebook is set up with a series of internal links from the table of contents.    
* All notebooks should have a table of contents which has the "Page", date, and title (information that allows the reader to understand your work).     
* Also, one of the perks of keeping all activities in a single document is that you can **search and find elements quickly**.     
* You can document anything you'd like, aside from logging your research activities. For example:
	* feel free to log all/any ideas for your research project([example](https://github.com/adnguyen/Notebooks_and_Protocols/blob/master/2016_notebook.md#page-39-2016-06-13-post-doc-project-idea-assessing-current-impacts-of-climate-change-in-natural-populations)) as an entry,     
	* or write down notes for a paper([example](https://github.com/adnguyen/Notebooks_and_Protocols/blob/master/2016_notebook.md#id-section36).      

* Lastly, you can share specific entries because of the three "#" automatically creates a link when the notebook renders on github.      


### Table of contents for 60 entries (Format is *Page: Date(with year-month-day). Title*)

## 2017-01-23

# Info update
* outline
* 20 min
* learning.engaging activity
* use board 
* take home messages
* examples from literature

* advances in sequencing tech
* range of applications: WGS, RNAseq, ChIPseq, targeted/capture seq (custom-designed probes)
	* reducing representation:
		* Use RNA (will be concentrated at highly expressed genes, such as actin)
		* GBS/RAD-seq
	* Illumina is generating 90% of global data
* library prep: depends on the application
	* extract DNA or RNA (cDNA synthesis)
	* fragment sample, size selection
	* ligate adapters
		* individual barcode
	* add on sequencing adapters
	* PCR
* Sequencing-by-Synthesis (SBS)
	* DNA with adapters get loaded onto a lane on a flow cell
	* DNA washed over the lane and attached to the oligos
	* bridge amplification -- cluster generation
	* incorporate labeled dNTP -- imaging and recording which nucleotide is added
* other technologies
	* PacBio
		* Single molecule, Real time
		* 10,000- 60,000bp
		* not as accurate
		* can be used for "haplotyping"
		* error rate 15%
* learning activity

Human genome project 
* Sanger (2001-2003) vs. Hiseq (2014)
	* 15 yr vs 1 day, 1 genome vs 45, $3 billion vs $1000

Choosing platforms:
* where is genetic variation, phenotypes of interest
* population vs individuals, comparative studies
* model organisms or not -- whether there is genetic information available
	* combining different approaches
	* de novo assembly for non-model
* demographic history
* Adaptive genetic variation
* gene expression variation

* length of reads
	* longer you have, assembly is easier, confidence is higher
* number of reads
* distribution
	* random or concentrated in an area?
	

Paper Discussion
* Ellegern et al. 

	We talked about:
	* reference genome
	
	
## 2017-01-25 

outline
* announcement
* info update: QTN
* debate
* discussion of group projects

# Glossary
* Reads
	* short 50bp
	* long 100, 150, 300bp (Miseq), 10,000-60,000bp (SMRT)
	* single vs paired-end	
	

  